webapp/public/javascripts/controllers.js

```diff
@@ -33,3 +24,17 @@ function UserController($scope, $http) {
         })
 
 }
+
+function RegisterController($scope, $http, $location) {
+
+    $scope.register = function(){
+        $http.post('/api/users/register', {email: $scope.email, password: $scope.password, firstname: $scope.firstname, lastname: $scope.lastname}).
+            success(function(data, status, headers, config) {
+
+                $location.path('/users', {firstname: $scope.firstname})
```

Comment:
Why are you passing "{firstname: $scope.firstname}"? I don't see anything in users.jade that might use this.